iceoryx_posh/include/iceoryx_posh/internal/roudi/introspection/fixed_size_container.hpp

```diff
@@ -50,8 +50,8 @@ class FixedSizeContainer
         if (nextElement >= 0)
         {
             m_freeIndex = nextElement;
-            m_values[m_freeIndex].value = element;
-            m_values[m_freeIndex].isValid = true;
+            m_values[static_cast<uint32_t>(m_freeIndex)].value = element;
```

Comment:
since capacity is `uint32_t` this is not a problem now, but it might be more robust to cast to `uint64_t` ot create a typedef e.g. `Capacity_t` and use that